src/main/java/seedu/address/model/task/Task.java

```diff
@@ -15,13 +17,15 @@
     private final String description;
     private final PriorityLevel priorityLevel;
     private boolean isCompleted;
+    private final Set<Milestone> milestoneSet;
```

Comment:
Setting milestoneSet to final means you cannot add new milestones to it, or change deadlines in the milestones